### ISSUE TITLE
Bug with computed property names in style objects

### ISSUE THREAD

**Issue by occar421** [stylelint/postcss-css-in-js#36]
> Clearly describe the bug

With styled-components Style Objects, we can nest the selector with [computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) like this.

```jsx
const Foo = styled.div({ color: "red" });
const Bar = styled.div({
  color: "white",
  [Foo]: { color: "green" },
});

// "hello" turns red, "world" turns green and others turn white
<>
  <Foo>hello</Foo>
  <Bar>css-in-js <Foo>world</Foo>!</Bar>
</>
```

postcss-css-in-js breaks the computed property names in the style objects.

> What steps are needed to reproduce the bug?

Just process that code with postcss with postcss-css-in-js syntax.

You can check it with https://github.com/occar421/demo-for-issue/tree/master/postcss-css-in-js--2.

> What did you expect to happen?

The output code will be the same and won't change the selector behavior and the style.

> What actually happened (e.g. what warnings or errors did you get)?

postcss-css-in-js syntax removes `[]` if the content is a simple variable ("Identifier"). In the example above, `[Foo]: {}` will be modified to `Foo: {}`, which changes the selector behavior.

**Comment by occar421** [comment]
I created a draft PR. (I didn't read the PR template until just before submitting it 😓)
